airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,856 @@
+    # [END gcp_transfer_operation_cancel_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 operation_name,
+                 api_version='v1',
+                 gcp_conn_id='google_cloud_default',
+                 *args,
+                 **kwargs):
+        super(GcpTransferServiceOperationCancelOperator, self).__init__(*args, **kwargs)
+        self.operation_name = operation_name
+        self.api_version = api_version
+        self.gcp_conn_id = gcp_conn_id
+        self._validate_inputs()
+
+    def _validate_inputs(self):
+        if not self.operation_name:
+            raise AirflowException("The required parameter 'operation_name' "
+                                   "is empty or None")
+
+    def execute(self, context):
+        hook = GCPTransferServiceHook(
+            api_version=self.api_version,
+            gcp_conn_id=self.gcp_conn_id,
+        )
+        hook.cancel_transfer_operation(
+            operation_name=self.operation_name
+        )
+
+
+class S3ToGoogleCloudStorageTransferOperator(BaseOperator):
+    """
+    Synchronizes an S3 bucket with a Google Cloud Storage bucket using the
+    GCP Storage Transfer Service.
+
+    .. warning::
+
+        This operator is NOT idempotent. If you run it many times, many transfer
+        job will be created on the Google Cloud Platform.
+
+    **Example**:
+
+    .. code-block:: python
+
+       s3_to_gcs_transfer_op = S3ToGoogleCloudStorageTransferOperator(
+            task_id='s3_to_gcs_transfer_example',
+            s3_bucket='my-s3-bucket',
+            project_id='my-gcp-project',
+            gcs_bucket='my-gcs-bucket',
+            dag=my_dag)
+
+    :param s3_bucket: The S3 bucket where to find the objects. (templated)
+    :type s3_bucket: str
+    :param gcs_bucket: The destination Google Cloud Storage bucket
+        where you want to store the files. (templated)
+    :type gcs_bucket: str
+    :param project_id: Optional ID of the Google Cloud Platform Console project that
+        owns the job
+    :type project_id: str
+    :param aws_conn_id: The source S3 connection
+    :type aws_conn_id: str
+    :param gcp_conn_id: The destination connection ID to use
+        when connecting to Google Cloud Storage.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param description: Optional transfer service job description
+    :type description: str
+    :param schedule: Optional transfer service schedule;
+        If not set, run transfer job once as soon as the operator runs
+        The format is described
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs.
+        With two additional improvements:
+
+        * dates they can be passed as :class:`datetime.date`
```

Comment:
In previous operators this section was missing the :class: directive. Might be worth taking a look at all of them and unifying them.

Reply:
I updated a previous description.